### PR TITLE
View user information feature

### LINKED ISSUE #21: View user information
Fully show most of the user information (i.e. finish the dashboard page)

### PR DESCRIPTION
**Pull request by andreeadutulescu**
View user information feature (#21)
Screenshot from 2021-05-10 10-58-14

### CODE REVIEW

File: src/component/Profile.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { Form, Button, Card, Alert } from 'react-bootstrap'
+import { Link, useHistory } from 'react-router-dom'
+
+import { useAuth } from '../contexts/AuthContext'
+import Client from '../models/Client'
+
+export function Profile() {
+  const { currentUser, updatePassword, updateEmail } = useAuth()
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
```
Comment (sebika) [inline src/component/Profile.js]: I think we don't need those here

File: src/component/Profile.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React, { useRef, useState } from 'react'
+import { Form, Button, Card, Alert } from 'react-bootstrap'
+import { Link, useHistory } from 'react-router-dom'
+
+import { useAuth } from '../contexts/AuthContext'
+import Client from '../models/Client'
+
+export function Profile() {
+  const { currentUser, updatePassword, updateEmail } = useAuth()
```
Comment (sebika) [inline src/component/Profile.js]: Here you are using only `currentUser`

File: src/component/Profile.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import React, { useRef, useState } from 'react'
+import { Form, Button, Card, Alert } from 'react-bootstrap'
+import { Link, useHistory } from 'react-router-dom'
+
+import { useAuth } from '../contexts/AuthContext'
+import Client from '../models/Client'
```
Comment (sebika) [inline src/component/Profile.js]: `Client` also is not used

File: src/component/Profile.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState(false)
+  const history = useHistory()
+
+  return (
+    <>
+      <Card>
+        <Card.Body>
+          <h2 className='text-center mb-4'>My Profile</h2>
+          {error && <Alert variant='danger'>{error}</Alert>}
```
Comment (sebika) [inline src/component/Profile.js]: We can't have an error because we are just viewing the information

File: src/component/Profile.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          {currentUser.db.get('mail')}</p>
+
+          <p><h6>National ID</h6>
+          {currentUser.db.get('CNP')}</p>
+
+          <p><h6>Address</h6>
+          {currentUser.db.get('adresa')}</p>
+
+          <Link to='/update-profile'>
+            <Button disabled={loading} className='w-100'>
```
Comment (sebika) [inline src/component/Profile.js]: `disabled` option is not necessary

**Review by sebika — CHANGES_REQUESTED** [review]

**Comment by sebika** [comment]
Let's try to add some styling similar to this one:
image

You can find reference in `/account-creation-requests`

File: src/component/Profile.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+              Update Profile
+            </Button>
+          </Link>
+        </Card.Body>
+      </Card>
+
```
Comment (Jokeswar) [inline src/component/Profile.js:39]: Can you also add a `Go back` button similar to the one found in the BankAccoutView?

**Review by Jokeswar — CHANGES_REQUESTED** [review]

**Comment by andreeadutulescu** [comment]
Fixed design
Screenshot from 2021-05-10 11-36-44

File: src/component/Profile.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+          <Link to='/update-profile'>
+            <Button className='w-100'>
+              Update Profile
+            </Button>
+          </Link>
+        </Card.Body>
+      </Card>
+
+      <div className='w-100 text-center mt-2'>
+        <Link to='/dashboard'>Go Back</Link>
```
Comment (Jokeswar) [inline src/component/Profile.js]: I was referring to  a button with `onClick={() => history.goBack()}`. It makes more sense this way. If you come from another page, different from `/dashboard`, `Go back` will not hold up to its meaning.

**Review by Jokeswar — CHANGES_REQUESTED** [review]

File: src/component/Profile.js (hunk 7 above)
Comment (sebika) [reply src/component/Profile.js]: @andreeadutulescu  you can use this:
```
<div className='w-100 text-center mt-2' style={{marginBottom:70}}>
  <Button variant='link' onClick={() => history.goBack()}>
    Go Back
  </Button>
</div>
```

**Review by sebika — APPROVED** [review]
Nice addition

**Review by Jokeswar — APPROVED** [review]

**Pull request merged**